src/net/sourceforge/kolmafia/textui/command/UmbrellaCommand.java

```diff
@@ -0,0 +1,66 @@
+package net.sourceforge.kolmafia.textui.command;
+
+import net.sourceforge.kolmafia.KoLCharacter;
+import net.sourceforge.kolmafia.KoLConstants.MafiaState;
+import net.sourceforge.kolmafia.KoLmafia;
+import net.sourceforge.kolmafia.RequestThread;
+import net.sourceforge.kolmafia.objectpool.ItemPool;
+import net.sourceforge.kolmafia.request.GenericRequest;
+import net.sourceforge.kolmafia.request.UmbrellaRequest;
+import net.sourceforge.kolmafia.session.InventoryManager;
+
+public class UmbrellaCommand extends AbstractCommand {
+  public UmbrellaCommand() {
+    this.usage =
+        "[ml | item | dr | weapon | spell | nc | broken | forward | bucket | pitchfork | twirling | cocoon] - fold your Umbrella";
+  }
+
+  @Override
+  public void run(final String cmd, String parameter) {
+    if (!InventoryManager.hasItem(ItemPool.UNBREAKABLE_UMBRELLA)
+        && !KoLCharacter.hasEquipped(ItemPool.UNBREAKABLE_UMBRELLA)) {
+      KoLmafia.updateDisplay(MafiaState.ERROR, "You need an Unbreakable Umbrella first.");
+      return;
+    }
+
+    parameter = parameter.trim();
+
+    if (parameter.equals("")) {
+      KoLmafia.updateDisplay(MafiaState.ERROR, "What state do you want to fold your umbrella to?");
+      return;
+    }
+
+    Integer umbrellaState = null;
+    if (parameter.equals("ml") || "broken".startsWith(parameter)) {
+      umbrellaState = UmbrellaRequest.Form.BROKEN.id;
+    } else if (parameter.equals("dr") || "forward".startsWith(parameter)) {
+      umbrellaState = UmbrellaRequest.Form.FORWARD.id;
+    } else if (parameter.equals("item") || "bucket".startsWith(parameter)) {
+      umbrellaState = UmbrellaRequest.Form.BUCKET.id;
+    } else if (parameter.equals("weapon") || "pitchfork".startsWith(parameter)) {
+      umbrellaState = UmbrellaRequest.Form.PITCHFORK.id;
+    } else if (parameter.equals("spell") || "twirling".startsWith(parameter)) {
+      umbrellaState = UmbrellaRequest.Form.TWIRL.id;
+    } else if (parameter.equals("nc") || "cocoon".startsWith(parameter)) {
+      umbrellaState = UmbrellaRequest.Form.COCOON.id;
+    }
+
+    if (umbrellaState == null) {
+      KoLmafia.updateDisplay(
+          MafiaState.ERROR, "I don't understand what Umbrella form " + parameter + " is.");
+      return;
+    }
```

Comment:
This is OK, but it could just be the "else" clause of the preceding if/elseif chain.

Reply:
I was copying the saber code, but it is cleaner looking to tag it onto the end there.